util/pkg/proxy/proxy_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package proxy
+
+import (
+	"reflect"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/kops/pkg/apis/kops"
+)
+
+func TestGetProxyEnvVars(t *testing.T) {
+	tests := []struct {
+		inProxies *kops.EgressProxySpec
+		expected  []v1.EnvVar
+	}{
+		{
+			inProxies: nil,
+			expected:  []v1.EnvVar{},
+		},
+		{
+			inProxies: &kops.EgressProxySpec{
+				HTTPProxy: kops.HTTPProxy{
+					Host: "",
+					Port: 1234,
+				},
+			},
+			// TODO: The output seems a little bit weird for this case
```

Comment:
We might want to catch it in validation, but this seems a reasonable translation at this layer!